package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -147,6 +235,16 @@ def __init__(self, atomgroup, **kwargs):
 
         phi_sel = [res.phi_selection() for res in residues]
         psi_sel = [res.psi_selection() for res in residues]
+        if any(sel is None for sel in phi_sel):
```

Comment:
This was added because `phi_selection()` and `psi_selection()` do not account for altloc selections. They will return `None` in the middle of the protein if that is the case, so this removes the `None` and the corresponding phi or psi angle. The amount of residues for which this is the case is very small so it seemed easier to just ignore them instead of rewriting `phi_selection()` and `psi_selection()`. If a user is interested in those specific residues, they can use the `Dihedral` class to look at them.

Reply:
Add a comment to the code, summarizing what you explained above.